src/safe_db/db.rs

```diff
@@ -31,6 +31,45 @@ impl DB {
 
     fn compute_edges(&mut self) {
         let mut edges = vec![];
+
+        // token address -> orga addresses
+        let mut organization_accepted_tokens: HashMap<Address, HashSet<Address>> = HashMap::new();
+
+        // Build a map from token address to orga addresses that accept this token
+        for (_, safe) in &self.safes {
+            for (send_to, percentage) in &safe.limit_percentage {
+                if percentage == &0 {
+                    continue;
+                }
+
+                let receiver_safe = self.safes.get(send_to).unwrap();
+                if receiver_safe.organization {
+                    //println!("user {} can send {} token to orga {}", user, safe.token_address, send_to);
+                    organization_accepted_tokens.entry(safe.token_address).or_default().insert(*send_to);
+                }
+            }
+        }
+
+        // Find all safes that have a non-zero balance of tokens that are accepted by an organization
+        for (user, safe) in &self.safes {
+            for (token, balance) in &safe.balances {
+                if balance == &U256::from(0) {
+                    continue;
+                }
+                organization_accepted_tokens.get(token).map(|organizations| {
```

Comment:
Does this compile? The function passed to `map` does not return anything.
Maybe `if let Some(organizations) = rganization_accepted_tokens.get(token) {`?

Reply:
or use something like `flat_map` and `for_each`

Reply:
`organization_accepted_tokens.get(token)` returns an `Option<HashSet<Address>>` so the method used is https://doc.rust-lang.org/std/option/enum.Option.html#method.map and not https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.map